File: Frontend/src/app/models/UserParams.ts
import { User } from "./User";

export class UserParams {
    gameType: string;
    minAge = 18;
    maxAge = 100;
    pageNumber = 1;
    pageSize = 12;

    constructor(user: User) {
        this.gameType = user.gameType;
        if(!user.gameType){
            this.gameType = "PVE & PVP";
        }
    }
}

File: Backend/Helpers/UserParams.cs
using System;
using System.Collections.Generic;
using System.Linq;
using System.Threading.Tasks;

namespace Backend.Helpers
{
    public class UserParams : PaginationParams
    {
        public string CurrentUsername { get; set; }
        public string GameType { get; set; }

        public int MinAge { get; set; } = 18;
        public int MaxAge { get; set; } = 200;

    }
}